### PR TITLE
Include Swift code into reference doc

### PR DESCRIPTION
**Pull request by spetrescu84**
## Proposed changes

- Changed build_docs.sh to add Swift code to reference doc using https://github.com/realm/jazzy#mixed-objective-c--swift
- A new branch was created which contains the latest script from https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1969 including all comments

## Type of change

- [x] Feature work
- [ ] Bug fix
- [x] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: build_docs.sh [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+
 echo -e "Generating MSAL documentation"
-jazzy --objc --umbrella-header MSAL/MSAL.h --framework-root . --sdk iphonesimulator --author Microsoft\ Corporation --author_url https://aka.ms/azuread  --github_url https://github.com/AzureAD/microsoft-authentication-library-for-objc --theme fullwidth
+# Generate Swift SourceKitten output
+sourcekitten doc -- -workspace MSAL.xcworkspace -scheme "MSAL (iOS Framework)" -configuration Debug RUN_CLANG_STATIC_ANALYZER=NO -sdk iphonesimulator CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -destination 'platform=iOS Simulator' > docs.temp/swiftDoc.json
+
+# Generate Objective-C SourceKitten output
+cd docs.temp
+sourcekitten doc --objc $(pwd)/MSAL/MSAL.h -- -x objective-c  -isysroot $(xcrun --show-sdk-path --sdk iphonesimulator) \-I $(pwd) -fmodules > objcDoc.json
+cd ..
```
Comment (nilo-ms) [inline build_docs.sh:21]: compared to the previous version of the script, we're moving to the parent directory. Is it expected?

File: build_docs.sh [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 
+# Feed both outputs to Jazzy as a comma-separated list
+jazzy --module MSAL --sourcekitten-sourcefile docs.temp/swiftDoc.json,docs.temp/objcDoc.json --author Microsoft\ Corporation --author_url https://aka.ms/azuread --github_url https://github.com/AzureAD/microsoft-authentication-library-for-objc --theme fullwidth
```
Comment (rodhan-ms) [inline build_docs.sh:24]: As this Jazzy command is different from the original, have you diff'ed the output of this script from before your changes with the output now to make sure the docs are the exact same (apart from the addition of the Native Auth parts of course)?

File: build_docs.sh (hunk 1 above)
Comment (spetrescu84) [reply build_docs.sh:21]: The script is the same as in previous PR and indeed different than the original one in this regard if that's what you meant - https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/9924950263e54dad10a426f42911f9044af27bfb/build_docs.sh and I've re-tested it. We have to run sourcekitten inside docs.tem but jazzy has to be run in the root directory as the xcodebuild (that jazzy does behind the scenes) has to be done from the root directory.

File: build_docs.sh (hunk 2 above)
Comment (spetrescu84) [reply build_docs.sh:24]: Yes, the files generated by Objective C are the same.

**Review by rodhan-ms — APPROVED** [review]

**Review by nilo-ms — APPROVED** [review]

**Review by antrix1989 — APPROVED** [review]

**Pull request merged**